e2e/specs/results-viewer.spec.ts

```diff
@@ -0,0 +1,168 @@
+    await page.locator('#ManualEntryAlkalinePhosphataseid').fill('10');
+    await page.locator('#ManualEntryAmylaseid').fill('10');
+    await page.locator('#ManualEntryBUNid').fill('10');
+    await page.locator('#manualEntryFastingBloodGlucosemgdlid').fill('10');
+    await page.locator('#manualEntryPostPrandialBloodGlucosemgdlid').fill('10');
+    await page.locator('#manualEntrySerumAlbuminid').fill('10');
+  });
+
+  await test.step('When I click the `Other` tab', async () => {
+    await page.getByRole('button', { name: 'Other', exact: true }).click();
+  });
+
+  await test.step('And I fill the "Urine" and "Stool" sections', async () => {
+    await page.locator('#manualEntryCultureandSensitivityUrineid').fill('83%');
+    await page.locator('#manualEntryUrinePregnancyTestid_1').check();
+    await page.locator('#manualEntryUrineProteinDipStickid_0').check();
+    await page.locator('#manualEntryUrineBacteriuriaTestid_1').check();
+    await page.locator('#manualEntryErythrocytesPresenceInUrineSedimentByLightMicroscopyTestid_1').check();
+    await page.locator('#manualEntryLeukocytesPresenceInUrineSedimentByLightMicroscopyid_2').check();
+    await page.locator('#manualEntryEpithelialCastsPresenceInUrineSedimentByLightMicroscopyTestid_2').check();
+    await page.locator('#manualEntryYeastPresenceInUrineSedimentByLightMicroscopyid_1').check();
+    await page.locator('#manualEntrySporePresenceInUrineTestid_3').check();
+    await page.locator('#manualEntryTrichomonasVaginalisPresenceInUrineSedimentByLightMicroscopyid_0').check();
+    await page.locator('#manualEntryCrystalsTypeInUrineSedimentByLightMicroscopyTestid_3').check();
+    await page.locator('#manualEntryStoolExamid').selectOption('Cestode Infection');
+    await page.locator('#manualEntryStoolFatTestSemiQuantitativeid_3').check();
+    await page.locator('#manualEntryStoolTestforReducingSubstanceid_0').check();
+    await page.locator('#manualEntryFecalOccultBloodTestid_2').check();
+  });
+
+  await test.step('And I click on the `Save and close` button', async () => {
+    await page.getByRole('button', { name: /save and close/i }).click();
+  });
+
+  await test.step('Then I should see a success toast notification', async () => {
+    await expect(page.getByText('The form has been submitted successfully.')).toBeVisible();
+  });
+
+  await test.step('When I go to the results viewer page', async () => {
+    await resultsViewerPage.goTo(patient.uuid);
+  });
+
+  await test.step('Then I should see the newly entered test results reflect in the results viewer', async () => {
+    await expect(page.getByRole('link', { name: /white blood cells/i })).toBeVisible();
+    await expect(page.getByText('12', { exact: true })).toBeVisible();
+    await expect(page.getByRole('link', { name: /red blood cells/i })).toBeVisible();
+    await expect(page.getByText('15', { exact: true })).toBeVisible();
+  });
+
+  await test.step('When I navigate to the `Visits` page', async () => {
```

Comment:
Any reason why we are navigating to the visits page then to the encounters? I think the motive is to assert that the lab results are reflected in the results viewer.

Reply:
@kdaud AFAIK, to edit lab results, you go through encounters and that's only available as a tab on the visits page.
Is there another approach, please?

Reply:
Ah I see, we definitely need this step :+1: